cmd/gateclient/client.go

```diff
@@ -400,6 +401,11 @@ func prompt() string {
 }
 
 func dialGate(gateClient *GatewayClient) error {
-	_, err := http.Get(gateClient.GateEndpoint())
-	return err
+	res, err := http.Get(gateClient.GateEndpoint())
+	if err != nil {
+		return err
+	}
+	io.Copy(ioutil.Discard, res.Body)
```

Comment:
I was unclear initially why this drain procedure was suggested in the original comment. Reading around it seems like the go http lib may not reuse connections unless the body is drained completely before closing. This incurs a cost when working with https endpoints because the TLS handshake then takes time to re-establish for subsequent requests.

Reply:
:+1: